### PR TITLE
add record's communities new resource stubs

### PR DESCRIPTION
**Pull request by ntarocco**
* adds a new resource /api/records/<pid>/communities
* creates stub methods
* extracts IIIF to a dedicated service
* extracts /api/communities/<pid>/records to a dedicated resource

### CODE REVIEW

**Review by ppanero — APPROVED** [review]
approving assuming tests will pass

File: invenio_rdm_records/resources/config.py [hunk 1]
```diff
@@ -228,6 +229,37 @@ class RDMParentRecordLinksResourceConfig(RecordResourceConfig, ConfiguratorMixin
     error_handlers = record_links_error_handlers
 
 
+#
+# Record's communities
+#
+class RDMRecordCommunitiesResourceConfig(RDMRecordResourceConfig, ConfiguratorMixin):
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply invenio_rdm_records/resources/config.py]: I worked together with Javier on this, we didn't have better naming in mind:
- Record's communities:
`/api/records/<pid>/communities`

- Community's records:
`/api/communities/<pid>/records`

**Review by jrcastro2 — APPROVED** [review]
Good starting point, thanks a lot!

**Review by zzacharo — APPROVED** [review]

File: invenio_rdm_records/resources/config.py [hunk 2]
```diff
@@ -230,6 +233,25 @@ class RDMParentRecordLinksResourceConfig(RecordResourceConfig, ConfiguratorMixin
     error_handlers = record_links_error_handlers
 
 
+#
+# Community's records
+#
+class RDMCommunityRecordsResourceConfig(RecordResourceConfig, ConfiguratorMixin):
```
Comment (zzacharo) [inline invenio_rdm_records/resources/config.py:239]: I just found out this https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/resources.py#L72. Isn't this similar to the resource that is added here?

**Pull request merged**